### ISSUE TITLE
Should the PlayPauseButton become disabled when the light source is off?

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/molecules-and-light#320]
The reasoning for doing this is to make it more clear that the photon emitter "On" button is the most important element in the sim, and make it so the PlayPauseButton isn't the first thing the user interacts with. But there are still some problematic cases with this feature that we should sort out. Before proceeding with #319 and making a change to common code in https://github.com/phetsims/scenery-phet/issues/592, it would be good to make sure we want to do this.

**Comment by jessegreenberg** [comment]
I added a query parameter `?disablePlayPause` to test this feature. It hacks into TimeControlNode so this is definitely not a long term solution, but it lets us try it out.

**Comment by jessegreenberg** [comment]
I think this was mentioned at the last design meeting, but this gif identifies one sequence of events with that is problematic. It creates a situation where photons are in the observation window but the user cannot resume play.
![ezgif com-video-to-gif](https://user-images.githubusercontent.com/6396244/77772421-7f94a800-701e-11ea-9fa1-588e45510ca4.gif)

@arouinfar @terracoda can you test out this version? Is this a change we should make? IF so, do you think the behavior in this gif acceptable?

**Comment by terracoda** [comment]
Good catch @jessegreenberg, I do see this as a problem especially since photons can be left in the observation window after pausing and then turning off the light source. For few Step Forward presses there are residual photons even though the light source is off.

I don't know why this hasn't come up before, but really, there are all kinds of UX problems and accessible UX problems with Time Control Node. 

The bigger question is how beneficial the Step Forward button is for this sim, especially since it now has two clear sim speeds "normal" and "slow."

Using UI Context Responses, I think we can address almost any strange edge case for non-visual users. In fact, UI Context Responses are already drafted if we do not change the Pause/Play at all.

That said, I am most interested in a sim that has interactions that are meant to be useful for as many learners as possible no matter what technology they use to access the sim. 

@arouinfar, do you think this interaction warrants Design Meeting time?

**Comment by arouinfar** [comment]
> I added a query parameter ?disablePlayPause

Thank you @jessegreenberg. I find the interaction very odd, and I don't think we want to be messing with the play/pause enabledProperty like this.

>  I don't know why this hasn't come up before, but really, there are all kinds of UX problems and accessible UX problems with Time Control Node.

I want to push back on this a bit. I have done dozens of interviews across many sims containing TimeControlNode, and I wouldn't say there are generally UX problems with this control. There have certainly been accessible UX problems within MAL, and we have made design changes which I think benefit all users (which really is the goal here).

There have been 10 interviews done on Molecules and Light with undergraduates. Eight of those were with a version containing a photon rate slider and play/pause/step. None of the play/pause/step interactions were problematic. Of those 8 interviews...

- 2/8 never used play/pause/step. 
- 2/8 briefly tested play/pause/step and identified their function after exploration had wrapped up. (One student said slow motion would be more useful).
- 2/8 briefly tested play/pause (but not step) and identified the function after exploration had wrapped up.
- 1/8 briefly used play/pause/step while exploring, but didn’t leverage it toward any specific goal.
- 1/8 used play/pause/step while exploring with the goal of slowing down the phenomena (slow motion would better support this)

The play/pause and step buttons are secondary controls and the instructions in a design interview are essentially "play around and see what you figure out",  so I'm not worried that most students didn't use the controls when exploring/making sense of the phenomena in the sim. 

I watched the clips from @BLFiedler's recent MAL interviews, and understand why there was some concern. I'm also hesitant to draw conclusions because the `emissionFrequency` was incredibly high, and I think fundamentally changes how students might interact with the TimeControlNode. 

> The bigger question is how beneficial the Step Forward button is for this sim, especially since it now has two clear sim speeds "normal" and "slow."

I think the `stepForwardButton` is still useful for those who really want to investigate the interaction frame-by-frame. It's still very much a secondary interaction, and many users may never use the step button, but that's okay with me.

> Using UI Context Responses, I think we can address almost any strange edge case for non-visual users. In fact, UI Context Responses are already drafted if we do not change the Pause/Play at all.

Great!

> That said, I am most interested in a sim that has interactions that are meant to be useful for as many learners as possible no matter what technology they use to access the sim.

Agreed! I think the changes we've made to MAL have really made the sim better for all users. My hesitation here is that the pre-1.0 interviews did not show any real problems with the `stepForwardButton`, and the high `emissionFrequency` in the latest round muddies the waters a bit. 

> @arouinfar, do you think this interaction warrants Design Meeting time?

Quite possibly, but let's see how the discussion goes during the description meeting on Tuesday.

**Comment by arouinfar** [comment]
In the MAL Meeting Agenda doc, @terracoda asks:
> Would it benefit many learners to remove the Pause/Play and Step Forward buttons all together?

There has never been a sim with Normal/Slow radio buttons, but no play/pause/step buttons. From user testing, we know that TimeControlNode as a whole is fairly intuitive across a variety of sims, but I do not think those findings necessarily generalize to a lone Normal/Slow radio button group. I would not publish a sim with just the normal/slow radio group without a fair amount of user testing. 

Use cases for play/pause/step:
* For most users, slow motion will be sufficient to watch the vibrating/rotating/dissociating. However, some users may still want to inspect the phenomena frame-by-frame.
* If the user's goal is to pause the sim at a precise moment, they are likely going to be off by a bit. The step button allows the user to advance to the precise moment they are wanting to inspect.
* The light source has a sound associated with it that will be at minimum 70BPM at normal speed. A pause may be useful to briefly cease the sound during discussions or at other times where it's not critical for the simulation to be playing. The alternative is to mute the sim altogether, but if you force users to mute, they are probably not going to unmute.

**Comment by terracoda** [comment]
Thanks for pointing out these use cases with details. I imagine the Pause and Step Forward would be very useful for teachers during demonstrations as well.

**Comment by terracoda** [comment]
@arouinfar, I really appreciate you taking the time to provide detail comments to my question.

**Comment by jessegreenberg** [comment]
This was discussed during design meeting today and after reviewing the points @arouinfar made above we decided not to disable the PlayPauseButton when the light source is off. Closing this issue.
